test/hotspot/jtreg/runtime/ErrorHandling/TestHeapDumpOnOutOfMemoryError.java

```diff
@@ -96,11 +97,25 @@ static void test(String type) throws Exception {
         output.stdoutShouldNotBeEmpty();
         output.shouldContain("Dumping heap to " + type + ".hprof");
         File dump = new File(heapdumpFilename);
-        Asserts.assertTrue(dump.exists() && dump.isFile(),
-                "Could not find dump file " + dump.getAbsolutePath());
-
+        Asserts.assertTrue(dump.exists() && dump.isFile(), "Could not find dump file " + dump.getAbsolutePath());
         HprofParser.parse(new File(heapdumpFilename));
-        System.out.println("PASSED");
+
+        // Test again using %p pid substitution in HeapDumpPath:
+        heapdumpFilename = type + ".%p.hprof";
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:+HeapDumpOnOutOfMemoryError",
+                "-XX:HeapDumpPath=" + heapdumpFilename,
+                "-XX:MaxMetaspaceSize=16m",
+                "-Xmx128m",
+                Platform.isDebugBuild() ? "-XX:-VerifyDependencies" : "-Dx",
+                TestHeapDumpOnOutOfMemoryError.class.getName(), type);
+
+        output = new OutputAnalyzer(pb.start());
+        output.stdoutShouldNotBeEmpty();
+        String actualHeapdumpFilename = type + "." + output.pid() + ".hprof";
+        output.shouldContain("Dumping heap to " + actualHeapdumpFilename);
```

Comment:
This better to be something like expectedlHeapdumpFilename and 
"Expected heap dump file".

Not very important, but make log cleaner.

Reply:
sure, done!